packages/lesswrong/components/thinkPage/ThinkSidePost.tsx

```diff
@@ -0,0 +1,74 @@
+// TODO: Import component in components.ts
+import React from 'react';
+import { registerComponent, Components } from '../../lib/vulcan-lib';
+import { ToCData, ToCSection } from '@/lib/tableOfContents';
+import { Link, useNavigate } from '@/lib/reactRouterWrapper';
+import { getSectionsWithOffsets, jumpToAnchorGeneric } from '../posts/TableOfContents/FixedPositionToC';
+import { useLocation } from '@/lib/routeUtil';
+
+const styles = (theme: ThemeType) => ({
+  root: {
+    padding: theme.spacing.unit * 2,
+    marginBottom: theme.spacing.unit,
+    borderRadius: 3,
+    marginTop: theme.spacing.unit * 1.5,
+    marginLeft: -6,
+    background: theme.palette.background.primaryTranslucent
+  },
+  section: {
+    marginTop: theme.spacing.unit * 1.5,
+    display: '-webkit-box',
+    maxHeight: '3em',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    WebkitLineClamp: 2, // Limit to two lines
+    WebkitBoxOrient: 'vertical',
+    lineHeight: '1.25em',
+  },
+  title: {
+    fontWeight: 'bold',
+  }
+});
+
+export const ThinkSidePost = ({classes, post, sectionData}: {
+  classes: ClassesType<typeof styles>,
+  post: PostsPage,
+  sectionData?: ToCData | null
+}) => {
+  const postContent = document.getElementById('postContent');
+  const newNormalizedSections = postContent ? getSectionsWithOffsets(postContent, sectionData?.sections ?? []) : sectionData?.sections ?? [];
```

Comment:
sorta hard to read, I would at a minimum format the ternary:
```
  const newNormalizedSections = postContent
    ? getSectionsWithOffsets(postContent, sectionData?.sections ?? [])
    : sectionData?.sections ?? [];
```

and probably also do:
```
  const sections = sectionData?.sections ?? [];
  const newNormalizedSections = postContent
    ? getSectionsWithOffsets(postContent, sections)
    : sections;
```